app/helpers/layout_helper.rb

```diff
@@ -91,7 +93,10 @@ def multiple_selects(f, attr, associations, selected_ids, options = {}, html_opt
       JSON.parse(html_options["data-useds"]).each do |disabled_value|
         hidden_fields += f.hidden_field(attr_ids, :multiple => true, :value => disabled_value, :id=>'' )
       end
-      hidden_fields + f.collection_select(attr_ids, associations.all.sort_by { |a| a.to_s },
+      unauthorized.each do |unauthorized_value|
+        hidden_fields += f.hidden_field(attr_ids, :multiple => true, :value => unauthorized_value, :id=>'' )
+      end
+      hidden_fields + f.collection_select(attr_ids, authorized.sort_by { |a| a.to_s },
```

Comment:
If the form does not send the unauthorized but selected values ( i.e. a subnet the user is not authorized to view but is associated with the location), then sending it will cause the association to be removed. 
The hidden fields contain all IDs of the items that must stay associated after submit - i.e. those that are used by hosts (which are disabled in the form and sent this way) and those that the user does not know exist.